### PR TITLE
React capstone project - Air Pollution Metrics web app

### PR DESCRIPTION
**Pull request by ahmidou09**
### General Requirements:

- Ensures a lint-error-free codebase.
- Adheres to the Gitflow branching strategy.
- Provides comprehensive and professional documentation.
- HTML/CSS & JavaScript Requirements:

### Follows best practices for HTML, CSS, and JavaScript development.

Implements responsive design following design guidelines, including color selection, typography, and layout inspired by Nelson Sakwa's original design.

### Project Requirements:

- Selects and integrates an API, providing numeric data based on user-defined parameters.
- Chooses an API that is open or requires an API key 
- Gives appropriate credit to the design author under the Creative Commons license in the project README.

Interactions:

### Home Page:

- Consuming the rest of countries v3.1 and the open weather map's air pollution APIs
- Enables navigation to the details page upon clicking a category item.

### Details Page:

- Users can interact with the page by getting a list of countries by region
- filter countries by names, and display detailed information about the pollution in the area
- air quality index bar that changes color depending on the level of pollution.

### Testing Requirements:

Creates unit tests for pure functions in Redux code.
Develop integration tests using React Testing Library, potentially mocking API access and Redux Store connections to ensure test integrity.

You can see a live demo here: [Live Demo](https://air-pollution-4yaz.onrender.com/)
Video explaining the features: [Video](https://drive.google.com/file/d/1tpDQ9bZgSqSHPTEakUpaPmjnGiAK5cJp/view?usp=sharing)

This pull request encompasses the development, documentation, and testing of the above project requirements, ensuring a robust and user-friendly web application.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,149 @@
...
+- [👥 Authors](#authors)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [📝 License](#license)
+
+<!-- PROJECT DESCRIPTION -->
+
+# 📖 REACT Bookstore <a name="about-project"></a>
+
```
Comment (AmaduKamara) [inline README.md]: - [x] For consistency in documenting your nicely developed app, kindly change the name from Bookstore to the actual app's project name. You must have missed that due to lots of work done. Please change that to have a proper balance on the documentation side 👍🏽

File: README.md [hunk 2]
```diff
@@ -1 +1,149 @@
...
+### Commands
+
+- ✔️ _npm start_:
+  Runs the app in the development mode.
+  
+- ✔️ _npm test_:
+  Launches the test runner in the interactive watch mode. See the section about running tests for more information.
+
+- ✔️ _npm run build_:
+  Builds the app for production to the build folder. It correctly bundles React in production mode and optimizes the build for the best performance.
```
Comment (AmaduKamara) [inline README.md]: - [x] Kindly add the command for installing the packages when the app is cloned. I believe that should be the first one before "npm start". Please add the `npm install` command at the top of the start command as that will not lead the user to have issues starting the app directly after cloning 👍🏽

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @ahmidou09,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ The app is hosted correctly and the local and live versions work fine as expected 👍🏽
✅ Your app is nicely designed 👍🏽

## Required Changes ♻️
_Check the comments under the review._
Good job so far for writing tests for the redux functions but,

- [x] It would be nice if you could add a few more tests for the react components using the React Testing Library as required in the project requirements.
- [x] Kindly improve on the title given to your PR. For this project, you could have titled your PR as "React capstone project - Air Pollution Metrics web app" as it's meaningful to the project title, and later inside the description of the PR, you'll then explain the tasks you worked on as you already did. 👍🏽

Screenshot 2023-09-06 at 8 21 57 PM

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by OLIPLICHE — APPROVED** [review]
Hi @ahmidou09, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Applied the changes requested by the previous  (✅)) 👏
- Created integrated test 👍 
- The test has passed 👏


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**